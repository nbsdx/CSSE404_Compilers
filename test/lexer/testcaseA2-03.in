public class GCD {
	public static int getGCD(int a, int b) {
		System.out.println((+ab+));
		if(b == 0)
			return a;
		int temp = ab;
		a = b;
		b = temp;
		return getGCD(a,b);
	}



