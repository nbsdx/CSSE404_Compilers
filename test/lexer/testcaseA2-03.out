ReservedWord, public
ReservedWord, class
ID, GCD
Delimiter, {
ReservedWord, public
ReservedWord, static
ReservedWord, int
ID, getGCD
Delimiter, (
ReservedWord, int
ID, a
Delimiter, ,
ReservedWord, int
ID, b
Delimiter, )
Delimiter, {
ReservedWord, System.out.println
Delimiter, (
Delimiter, (
Operator, +
ID, ab
Operator, +
Delimiter, )
Delimiter, )
Delimiter, ;
ReservedWord, if
Delimiter, (
ID, b
Operator, ==
Integer, 0
Delimiter, )
ReservedWord, return
ID, a
Delimiter, ;
ReservedWord, int
ID, temp
Delimiter, =
ID, ab
Delimiter, ;
ID, a
Delimiter, =
ID, b
Delimiter, ;
ID, b
Delimiter, =
ID, temp
Delimiter, ;
ReservedWord, return
ID, getGCD
Delimiter, (
ID, a
Delimiter, ,
ID, b
Delimiter, )
Delimiter, ;
Delimiter, }
